src/components/posts/utils/getAllPosts.js

```diff
@@ -0,0 +1,37 @@
+import client from "../../../utils/client";
+
+const secOneMin = 60;
+const secOneHour = secOneMin * 60;
+const secOneDay = secOneHour * 24;
+
+function formatTime(timeString) {
+  const time = new Date(timeString);
+  const now = new Date();
+  const diff = (now - time) / 1000;
+  const timeArr = time.toUTCString().split(" ");
+  let formatted;
+
+  if (time.getFullYear < now.getFullYear()) {
+    formatted = timeArr.slice(0, 3).join(" ");
+  } else if (diff > secOneDay) {
+    formatted = timeArr.slice(0, 2).join(" ");
+  } else if (diff > secOneHour) {
+    const hoursPassed = Math.floor(diff / secOneHour);
+    formatted = hoursPassed + " hours ago";
+  } else if (diff > secOneMin) {
+    const minutesPassed = Math.floor(diff / secOneMin);
+    formatted = minutesPassed + " minutes ago";
+  }
+
+  return formatted;
+}
+
+export async function getAllPostsFormatted() {
+  await client.get("/posts").then((res) => {
+    const posts = res.data.data.posts;
+    const formattedPosts = posts.map(
+      (post) => (post.createdAt = formatTime(post.createdAt))
+    );
+    return formattedPosts;
+  });
```

Comment:
`await client.get("/posts").then`

This won't work. You either need to use `await` *or* `.then` to handle promises, I suspect the component is failing to render properly because of this